overlord/snapstate/backend_test.go

```diff
@@ -676,9 +676,12 @@ func (f *fakeSnappyBackend) SetupSnap(snapFilePath, instanceName string, si *sna
 		snapType = snap.TypeGadget
 	}
 	if instanceName == "borken-in-setup" {
-		return snapType, fmt.Errorf("cannot install snap %q", instanceName)
+		return snapType, nil, fmt.Errorf("cannot install snap %q", instanceName)
 	}
-	return snapType, nil
+	if instanceName == "some-snap-no-install-record" {
+		return snapType, nil, nil
+	}
+	return snapType, &backend.InstallRecord{}, nil
```

Comment:
I'm not sure where it makes the most sense to put the test, but IMHO it would be good to have something in the fake backend like 

https://github.com/snapcore/snapd/blob/f7426b70ac1ed3d2375b17c5fcbb4c8afadab650/overlord/snapstate/backend_test.go#L805-L811

in the method so that we can mock the backend reporting a symlink being there and ensure that the manager calling the backend does the right thing

Reply:
I will play with both suggestions (yours and Samuele's) in a followup, I hope that's ok with you, thanks!